### PR TITLE
Implement Error on our error types

### PR DESCRIPTION
**Pull request by sgwilym**
I was pleasantly surprised by how few of these there seem to be. Would like review of clarity of display messages.

### CODE REVIEW

File: data-model/src/entry.rs [hunk 1]
```diff
@@ -212,7 +212,19 @@ where
 #[derive(Debug)]
 pub struct UnauthorisedWriteError;
 
-/// An AuthorisedEntry is a pair of an [`Entry`] and [`AuthorisationToken`](https://willowprotocol.org/specs/data-model/index.html#AuthorisationToken) (willowprotocol.org) implementing [`AuthorisationToken`] for which [`is_authorised_write`](https://willowprotocol.org/specs/data-model/index.html#is_authorised_write) returns true.
+impl core::fmt::Display for UnauthorisedWriteError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Tried to authorise the writing of an entry using a token which does not permit it."
```
Comment (AljoschaMeyer) [inline data-model/src/entry.rs:219]:
```suggestion
            "Tried to authorise the writing of an entry using an AuthorisationToken which does not permit it."
```

File: meadowcap/src/lib.rs [hunk 2]
```diff
@@ -136,6 +136,35 @@ pub enum FailedDelegationError<
...
+            FailedDelegationError::AreaNotIncluded {
+                excluded_area: _,
+                claimed_receiver: _,
+            } => write!(
+                f,
+                "Tried to delegate access to an area not fully included by the granted area of the capability being delegated from."
+            ),
+            FailedDelegationError::WrongSecretForUser(_) => write!(
+                f,
+                "Provided the wrong secret for the capability's receiver."
```
Comment (AljoschaMeyer) [inline meadowcap/src/lib.rs:153]: This sounds like you need the secret key of Alfie in order to delegate to Alfie.

File: meadowcap/src/lib.rs [hunk 3]
```diff
@@ -158,6 +187,45 @@ pub enum InvalidDelegationError<
...
+    > core::fmt::Display for InvalidDelegationError<MCL, MCC, MPL, UserPublicKey, UserSignature>
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            InvalidDelegationError::AreaNotIncluded {
+                excluded_area: _,
+                claimed_receiver: _,
+            } => write!(
+                f,
+                "Tried to append a delegation which granted access to an area not fully included by the granted area of the capability."
```
Comment (AljoschaMeyer) [inline meadowcap/src/lib.rs:205]:
```suggestion
                "Tried to append a delegation which grants access to an area not fully included by the granted area of the capability."
```

File: meadowcap/src/lib.rs [hunk 4]
```diff
@@ -158,6 +187,45 @@ pub enum InvalidDelegationError<
...
+            ),
+            InvalidDelegationError::InvalidSignature {
+                expected_signatory: _,
+                claimed_receiver: _,
+                signature: _,
+            } => write!(
+                f,
+                "Tried to append a delegation with an invalid signature for the receiver of the target capability."
```
Comment (AljoschaMeyer) [inline meadowcap/src/lib.rs:213]: This sounds like the receiver has to issue the signature.

File: meadowcap/src/lib.rs (hunk 2 above)
Comment (sgwilym) [reply meadowcap/src/lib.rs:153]: This is written from the perspective of the capability prior to being delegated, where the receiver is the one who delegates, which seems natural to me.

File: meadowcap/src/lib.rs (hunk 4 above)
Comment (sgwilym) [reply meadowcap/src/lib.rs:213]: Same thing as https://github.com/earthstar-project/willow-rs/pull/39#discussion_r1712615478

File: meadowcap/src/lib.rs (hunk 2 above)
Comment (sgwilym) [reply meadowcap/src/lib.rs:153]: As in, I am talking about the *actual* receiver of the capability, not the future capability if this operation were to succeed.

Comment (AljoschaMeyer) [reply meadowcap/src/lib.rs:153]: Still couldn't hurt to clarify this, I guess.

**Pull request merged**